test/jdk/javax/swing/JButton/SwingButtonResizeTestWithOpenGL.java

```diff
@@ -0,0 +1,311 @@
+            public void focusGained(FocusEvent fe) {
+                focusGainedLatch.countDown();
+            }
+        });
+        frame.getContentPane().setLayout(new FlowLayout());
+        frame.getContentPane().add(button);
+        frame.pack();
+        frame.setVisible(true);
+        frame.toFront();
+    }
+
+    public static void main(String[] args) throws Exception {
+        focusGainedLatch = new CountDownLatch(1);
+        SwingButtonResizeTestWithOpenGL test = new SwingButtonResizeTestWithOpenGL();
+        test.runTest();
+    }
+
+    public void runTest() throws Exception {
+        BufferedImage bimage1;
+        BufferedImage bimage2;
+
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+            robot.setAutoDelay(200);
+
+            if (focusGainedLatch.await(3, TimeUnit.SECONDS)) {
+                System.out.println("Button focus gained...");
+            } else {
+                System.out.println("Button focus not gained...");
+                throw new RuntimeException("Can't gain focus on button even after waiting too long..");
+            }
+
+            System.out.println("Getting initial button image..image1");
+            bimage1 = getButtonImage();
+
+            File file1 = new File("image1.png");
+            saveButtonImage(bimage1, file1);
+
+            //some platforms may not support maximize frame
+            if (frame.getToolkit().isFrameStateSupported(JFrame.MAXIMIZED_BOTH)) {
+
+                robot.waitForIdle();
+
+                frame.setExtendedState(JFrame.MAXIMIZED_BOTH);  //maximize
+                // frame from normal size
+                System.out.println("Frame is maximized");
+                robot.waitForIdle();
+
+                if (frame.getToolkit().isFrameStateSupported(JFrame.NORMAL)) {
+                    System.out.println("Frame is back to normal");
+                    frame.setExtendedState(JFrame.NORMAL);  //resize from
+                    // maximum size to normal
+
+                    //capture image of JButton after resize
+                    System.out.println("Getting image of JButton after resize" + "..image2");
+                    bimage2 = getButtonImage();
+                    File file2 = new File("image2.png");
+                    saveButtonImage(bimage2, file2);
+
+                    //compare button images from before and after frame resize
+                    DiffImage di = new DiffImage(bimage1.getWidth(), bimage1.getHeight());
+                    di.compare(bimage1, bimage2);
+                    System.out.println("Taking the diff of two images, image1" + " and image2");
+                    //results of image comparison
+                    int numDiffPixels = di.getNumDiffPixels();
+                    if (numDiffPixels != 0) {
+                        throw new RuntimeException("Button renderings are " + "different after window resize," + " num of Diff Pixels=" + numDiffPixels);
```

Comment:
Why are these prints composed of multiple strings with `+` joining them instead of one continuous string sentence? Seems a little odd and is seen multiple times throughout the test.

```suggestion
                    System.out.println("Getting image of JButton after resize..image2");
                    bimage2 = getButtonImage();
                    File file2 = new File("image2.png");
                    saveButtonImage(bimage2, file2);

                    // compare button images from before and after frame resize
                    DiffImage di = new DiffImage(bimage1.getWidth(), bimage1.getHeight());
                    di.compare(bimage1, bimage2);
                    System.out.println("Taking the diff of two images, image1 and image2");
                    // results of image comparison
                    int numDiffPixels = di.getNumDiffPixels();
                    if (numDiffPixels != 0) {
                        throw new RuntimeException("Button renderings are different after window resize, num of Diff Pixels = " + numDiffPixels);
```